src/main/java/com/mraof/minestuck/world/storage/MSExtraData.java

```diff
@@ -181,6 +211,43 @@ public void recoverConnections(Consumer<EditData.ConnectionRecovery> recover)
 		}
 	}
 	
+	/**
+	 * Gets the places the editmode server player can be for their client
+	 */
+	public EditmodeLocations getEditmodeLocations(UUID playerID)
```

Comment:
Is the player id meant to be for the client player? Should the editmode locations instance be tied to a specific sburb connection? The choice of player id used doesn't seem to be consistent at the moment. This is pretty important for reviewing the rest of the PR.

Also, it looks like this can return null, and should be annotated as such.

Reply:
I wasnt sure if it was smart to have it related to a connection with the idea that the locations shouldnt need to be remade if a connection is changed or broken

Reply:
Sounds like you might have an example scenario in mind?
Also if you could answer the player id question too that'd be nice.

Reply:
Not sure what I was necessarily thinking, but I decided after all to store the locations inside SburbConnection, which now works successfully